src/Microsoft.AspNetCore.Server.Kestrel/Infrastructure/TaskUtilities.cs

```diff
@@ -26,12 +26,15 @@ public static Task GetCancelledTask(CancellationToken cancellationToken)
 #endif
         }
 
-        public static Task<int> GetCancelledZeroTask()
+        public static Task<int> GetCancelledZeroTask(CancellationToken cancellationToken)
```

Comment:
How about making it an optional parameter with a default value of `default(CancellationToken)`?